ballerina/http_connection.bal

```diff
@@ -138,11 +138,12 @@ public isolated client class Caller {
             }
         } else if message is error {
             if message is ApplicationResponseError {
+                string? retrievedMediaType = retrieveMediaTypeForErrorResponse(returnMediaType, message.detail().get(CONTENT_TYPE));
```

Comment:
error detail does not contain a field called CONTENT_TYPE. You need to access the headers

Reply:
Fixed with [28f2606e58a20d925a1569c3070c9fe670884718](https://github.com/ballerina-platform/module-ballerina-http/pull/839/commits/28f2606e58a20d925a1569c3070c9fe670884718)